### PR TITLE
Support for sending private messages to reviewers

### PR DESCRIPTION
**Pull request by johnallen3d**
Not all users are subscribed to notifications from GitHub. This is taking a crack at #43.

### CODE REVIEW

File: lib/lita/handlers/reviewme.rb [hunk 1]
```diff
@@ -136,6 +137,17 @@ def get_room(response)
...
 
+      def all_reviewers(room)
+        ns_redis(room.id).lrange(REDIS_LIST, 0, -1)
+      end
+
+      def notify(reviewer, url)
+        return unless reviewer.chat_handle
+
+        target = Lita::Source.new(user: reviewer.chat_handle)
+        robot.send_message(target, "Hi there, you've been assigned to review #{url}.")
```
Comment (johnallen3d) [inline lib/lita/handlers/reviewme.rb]: Turns out this doesn't work they way I thought. Will require a little more work.

**Comment by iamvery** [comment]
I'm going to be out next week teaching. I'll try to get to this when soon, but there will be a little delay. Just wanted to let you know I see it :heart:

**Comment by iamvery** [comment]
It's not clear to me at a glance. Is adding a GH name optional? That is, does the plugin still behave as it does currently, allowing reviewers to be added without the GH name for notification?

Honestly, I think I'd rather have a separate mechanism (outside of lita-reviewme) for associating a GH handle to a Lita user. And then maybe hook into that to ping folks or something. Essentially, I'm not sure there is a ton of value to this feature at the cost of adding more complexity to the handler.

What are you thoughts?

**Comment by killpack** [comment]
@iamvery: it looks like this PR is adding support for private messaging a user when they've been assigned a review.

Currently, the plugin only allows you to assign a reviewer a GH name. This PR lets you assign a reviewer GH name and also, optionally, a chat handle. If you have an associated chat handle, you'll get both a comment on the PR with your GH name *and* a ping in Slack (or whatever). If you don't have a chat handle, the plugin works like it always has.

Example:
If you'd been set up like so: `lita add reviewer iamvery:jay` 
...then when you're assigned to review stuff, Lita would post ":eyes: @iamvery" on a GH PR and send a private message to `jay` in Slack.

**Comment by johnallen3d** [comment]
@iamvery, @killpack has hit the nail on the head. The plugin still behaves the same, support for associating a chat handle with a GitHub handle is optional. That being said I can see your point, having that association as part of this plugin might be overkill. Since developing this I've added my GitHub username as a [highlight word in Slack](https://github.com/iamvery/lita-reviewme/issues/43#issuecomment-188368341) and that's meeting my needs (can't speak for other platforms like HipChat or Campfire).

Anyhoo, feel free to reject this PR if it's overcomplicating the codebase.

**Comment by iamvery** [comment]
Thanks for the reply @johnallen3d. I don't want to make things hard to contribute, so I struggle with "no". I'm glad you understand where I'm coming from. It feels to be that it adds some unnecessary complexity and perhaps oversteps the handler's responsibility.

In my ideal world, the GH handle would be associated with the user via some-other-thing and you add/remove reviewers by the chat handle. Something like

> bot: add reviewer @jay
> bot: done!
> bot: review https://github.com/some/thing
> # on github...
> # :eyes: @iamvery

That seems better, right? And the stop gap works well until we can pull off something like that.

**Comment by iamvery** [comment]
Actually @sebastiangeiger might be onto something here https://github.com/sebastiangeiger/lita-identity-manager

**Comment by johnallen3d** [comment]
That makes a lot of sense. Having done very little `lita` development I'm not sure, is there a prescribed way for handlers to share data?

**Comment by iamvery** [comment]
I'm not sure myself. However, I believe that lita user's may have metadata (http://www.rubydoc.info/gems/lita/Lita/User#metadata-instance_method), so my thinking is take advantage of that interface.

**Pull request closed without merging**